### PR TITLE
[MNG-6931] Deprecate DefaultLog

### PR DESCRIPTION
**Pull request by michael-o**

### CODE REVIEW

**Review by mthmulders — APPROVED** [review]
Happy to see this change! Also happy to see that the migration path is clearly documented. LGTM.

File: maven-plugin-api/src/main/java/org/apache/maven/monitor/logging/DefaultLog.java [hunk 1]
```diff
@@ -24,7 +24,10 @@
 
 /**
  * @author jdcasey
+ *
+ * @deprecated Use SFL4J directly
```
Comment (elharo) [inline maven-plugin-api/src/main/java/org/apache/maven/monitor/logging/DefaultLog.java:28]: SLF4J

File: maven-plugin-api/src/main/java/org/apache/maven/plugin/AbstractMojo.java [hunk 2]
```diff
@@ -167,7 +171,9 @@ public void setLog( Log log )
      * method directly whenever you need the logger, it is fast enough and needs no caching.
      *
      * @see org.apache.maven.plugin.Mojo#getLog()
+     * @deprecated Use SFL4J directly
```
Comment (elharo) [inline maven-plugin-api/src/main/java/org/apache/maven/plugin/AbstractMojo.java:174]: SLF4J

File: maven-plugin-api/src/main/java/org/apache/maven/plugin/AbstractMojo.java [hunk 3]
```diff
@@ -149,6 +149,10 @@
     /** Plugin container context */
     private Map pluginContext;
 
+    /**
+     * @deprecated Use SFL4J directly
```
Comment (elharo) [inline maven-plugin-api/src/main/java/org/apache/maven/plugin/AbstractMojo.java:153]: SLF4J

File: maven-plugin-api/src/main/java/org/apache/maven/plugin/Mojo.java [hunk 4]
```diff
@@ -54,8 +54,10 @@ void execute()
      * and feedback to the user.
      *
      * @param log a new logger
+     *
+     * @deprecated Use SFL4J directly
```
Comment (elharo) [inline maven-plugin-api/src/main/java/org/apache/maven/plugin/Mojo.java:58]: SLF4J

File: maven-plugin-api/src/main/java/org/apache/maven/plugin/Mojo.java [hunk 5]
```diff
@@ -64,6 +66,8 @@ void execute()
      * @return a log4j-like logger object which allows plugins to create messages at levels of <code>"debug"</code>,
      * <code>"info"</code>, <code>"warn"</code>, and <code>"error"</code>. This logger is the accepted means to display
      * information to the user.
+     * @deprecated Use SFL4J directly
```
Comment (elharo) [inline maven-plugin-api/src/main/java/org/apache/maven/plugin/Mojo.java:69]: use SLF4J

File: maven-plugin-api/src/main/java/org/apache/maven/plugin/Mojo.java [hunk 6]
```diff
@@ -64,6 +66,8 @@ void execute()
      * @return a log4j-like logger object which allows plugins to create messages at levels of <code>"debug"</code>,
      * <code>"info"</code>, <code>"warn"</code>, and <code>"error"</code>. This logger is the accepted means to display
```
Comment (elharo) [inline maven-plugin-api/src/main/java/org/apache/maven/plugin/Mojo.java:67]: delete "This logger...to the user."

File: maven-plugin-api/src/main/java/org/apache/maven/plugin/AbstractMojo.java (hunk 2 above)
Comment (elharo) [inline maven-plugin-api/src/main/java/org/apache/maven/plugin/AbstractMojo.java:174]: use

**Review by elharo — COMMENTED** [review]
just some nits

File: maven-plugin-api/src/main/java/org/apache/maven/monitor/logging/DefaultLog.java (hunk 1 above)
Comment (michael-o) [reply maven-plugin-api/src/main/java/org/apache/maven/monitor/logging/DefaultLog.java:28]: Blind me, copy and paste error.

**Comment by slachiewicz** [comment]
What about deprecating also org.apache.maven.plugin.logging.Log ?

**Comment by michael-o** [comment]
> Log

This perfectly makes sense.

**Comment by michael-o** [comment]
> 
> 
> What about deprecating also org.apache.maven.plugin.logging.Log ?

Done.

**Review by slachiewicz — APPROVED** [review]

**Review by elharo — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #8409

**Pull request closed without merging**